src/server/req_runjob.c

```diff
@@ -484,66 +478,36 @@ req_runjob(struct batch_request *preq)
 	/* OK - go back over the run job request, assign the vhosts */
 	/* and finally run the job by calling req_runjob2()	    */
 
-	if (jt == IS_ARRAY_NO) {
+	if (jt == IS_ARRAY_Single) {
 
-		/* just a regular job, pass it on down the line and be done */
-		if (call_to_process_hooks(preq, hook_msg, sizeof(hook_msg),
-				pbs_python_set_interrupt) == 0) {
-			reply_text(preq, PBSE_HOOKERROR, hook_msg);
-			return;
-		}
-		pjob = where_to_runjob(preq, parent);
-		if (pjob) {
-			/* free prov_vnode before use */
-			job_attr_def[(int)JOB_ATR_prov_vnode].at_free(
-				&pjob->ji_wattr[(int)JOB_ATR_prov_vnode]);
-			req_runjob2(preq, parent);
-		}
-		return;
+	/* single subjob, if parent qeueud, it can be run */
 
-	} else if (jt == IS_ARRAY_Single) {
-
-		/* single subjob, if parent qeueud, it can be run */
-
-		offset = subjob_index_to_offset(parent, get_index_from_jid(jid));
-		if (offset == -1) {
-			req_reject(PBSE_UNKJOBID, 0, preq);
-			return;
-		}
-		i = get_subjob_state(parent, offset);
-		if (i == -1) {
-			req_reject(PBSE_IVALREQ, 0, preq);
-			return;
-		} else if ((i != JOB_STATE_QUEUED) || (find_job(jid) != NULL)) {
-			/* job already running */
-			req_reject(PBSE_BADSTATE, 0, preq);
-			return;
-		}
 		if ((pjobsub = create_subjob(parent, jid, &j)) == NULL) {
 			req_reject(j, 0, preq);
 			return;
 		}
+		pjob = pjobsub;
+	}
+
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
+
+		/* just a regular job or a subjob, pass it on down the line and be done */
 		if (call_to_process_hooks(preq, hook_msg, sizeof(hook_msg),
 				pbs_python_set_interrupt) == 0) {
 			reply_text(preq, PBSE_HOOKERROR, hook_msg);
 			return;
 		}
-		pjob = where_to_runjob(preq, pjobsub);
-		if (pjob == NULL) {
-			/* requeue subjob */
-			pjobsub->ji_qs.ji_substate = JOB_SUBSTATE_RERUN3;
-			job_purge(pjobsub);
-			return;
+		pjob = where_to_runjob(preq, pjob);
```

Comment:
What happends if where_to_runjob() returns NULL. Will the subjob be requeued which was the case earlier ?

Reply:
Fixed in rc 9 https://github.com/PBSPro/pbspro/pull/590/commits/81a72a19d9877270ee663e2df46bc9d46538a676